src/concurrency.md

```diff
@@ -73,6 +74,63 @@ fn main() {
 }
 ```
 
+[ex-rayon-map-reduce]: #ex-rayon-map-reduce
+<a name="ex-rayon-map-reduce"></a>
+## Map-reduce in parallel
+
+[![rayon-badge]][rayon] [![cat-concurrency-badge]][cat-concurrency]
+
+This example uses [`rayon::filter`], [`rayon::map`], and [`rayon::reduce`]
+to calculate the coniditional average age of a vector of `Person` objects.
+
+[`rayon::filter`] allows us to (in parallel) conditionally include elements from
+a collection that satisfy the given predicate.  Similarly, [`rayon::map`] and
+[`rayon::reduce`] allow us to transform the filtered elements via a unary
+operation and reduce them to a single value via a given binary operation,
+respectively.
+
+```rust
+extern crate rayon;
+
+use rayon::prelude::*;
+
+struct Person {
+    age: u32,
+}
+
+fn main() {
+    let v: Vec<Person> = vec![
+        Person { age: 23 },
+        Person { age: 19 },
+        Person { age: 42 },
+        Person { age: 17 },
+        Person { age: 17 },
+        Person { age: 31 },
+        Person { age: 30 },
+    ];
+
+    let gt_30 = |x: u32| x > 30;
+    let num_over_30 = v.par_iter().filter(|&ref x| gt_30(x.age)).count();
+    let sum_over_30 = v.par_iter()
+        .filter(|&ref x| gt_30(x.age))
+        .map(|ref x| x.age)
```

Comment:
I'd suggest mapping first and then using `filter(gt_30)` instead of closure in closure